ibis/backends/sql/__init__.py

```diff
@@ -413,3 +454,60 @@ def _compile_pandas_udf(self, udf_node: ops.ScalarUDF) -> str:
         raise NotImplementedError(
             f"pandas UDFs are not supported in the {self.name} backend"
         )
+
+    def _to_catalog_db_tuple(self, table_loc: sge.Table):
+        if table_loc is None or table_loc == (None, None):
+            return None, None
+
+        if (sg_cat := table_loc.args["catalog"]) is not None:
+            sg_cat.args["quoted"] = False
+            sg_cat = sg_cat.sql(self.name)
+        if (sg_db := table_loc.args["db"]) is not None:
+            sg_db.args["quoted"] = False
+            sg_db = sg_db.sql(self.name)
+
+        return sg_cat, sg_db
+
+    def _to_sqlglot_table(self, database):
+        if database is None:
+            return None
+        elif isinstance(database, tuple):
+            if len(database) > 2:
+                raise ValueError(
+                    "Only database hierarchies of two or fewer levels are supported."
+                    "\nYou can specify ('catalog', 'database')."
+                )
+            elif len(database) == 2:
+                catalog, database = database
+            elif len(database) == 1:
+                database = database[0]
+                catalog = None
+            else:
+                raise ValueError(
+                    f"Malformed database tuple {database} provided"
+                    "\nPlease specify one of:"
+                    '\n("catalog", "database")'
+                    '\n("database",)'
+                )
+            database = sg.exp.Table(
+                catalog=sg.to_identifier(catalog, quoted=self.compiler.quoted),
+                db=sg.to_identifier(database, quoted=self.compiler.quoted),
+            )
+        elif isinstance(database, str):
+            # There is no definition of a sqlglot catalog.database hierarchy outside
+            # of the standard table expression.
+            # sqlglot parsing of the string will assume that it's a Table
+            # so we unpack the arguments into a new sqlglot object, switching
+            # table (this) -> database (db) and database (db) -> catalog
+            table = sg.parse_one(database, into=sg.exp.Table, dialect=self.dialect)
+            if table.args["catalog"] is not None:
+                raise exc.IbisInputError(
+                    f"Overspecified table hierarchy provided: `{table.sql(self.dialect)}`"
+                )
+            catalog = table.args["db"]
+            db = table.args["this"]
+            database = sg.exp.Table(catalog=catalog, db=db)
+        else:
+            raise ValueError("oops")
+
+        return database
```

Comment:
Sorry you had to write this :(

Reply:
yeah, me too, but I think we'll get some mileage out of it.  This can also help with future usage patterns like `con.table("catalog.database.table")`